libafl/src/executors/forkserver.rs

```diff
@@ -417,21 +420,23 @@ where
 
         let last_run_timed_out = self.executor.forkserver().last_run_timed_out();
 
-        match &mut self.executor.shmem_mut() {
-            Some(shmem) => {
-                let target_bytes = input.target_bytes();
-                let size = target_bytes.as_slice().len();
-                let size_in_bytes = size.to_ne_bytes();
-                // The first four bytes tells the size of the shmem.
-                shmem.as_mut_slice()[..4].copy_from_slice(&size_in_bytes[..4]);
-                shmem.as_mut_slice()[SHMEM_FUZZ_HDR_SIZE..(SHMEM_FUZZ_HDR_SIZE + size)]
-                    .copy_from_slice(target_bytes.as_slice());
-            }
-            None => {
-                self.executor
-                    .input_file_mut()
-                    .write_buf(input.target_bytes().as_slice())?;
+        if self.executor.uses_shmem_testcase() {
+            match &mut self.executor.shmem_mut() {
+                Some(shmem) => {
+                    let target_bytes = input.target_bytes();
+                    let size = target_bytes.as_slice().len();
+                    let size_in_bytes = size.to_ne_bytes();
+                    // The first four bytes tells the size of the shmem.
+                    shmem.as_mut_slice()[..4].copy_from_slice(&size_in_bytes[..4]);
+                    shmem.as_mut_slice()[SHMEM_FUZZ_HDR_SIZE..(SHMEM_FUZZ_HDR_SIZE + size)]
+                        .copy_from_slice(target_bytes.as_slice());
+                }
+                None => {}
```

Comment:
Why not unwrap?

Reply:
We can even unwrap_unchecked here, right?

Reply:
ah that code side escaped me

Reply:
ah didn't see your second comment in time but yes I guess we could.
Pushed.

Reply:
Sweet